### PR TITLE
Add more check onGetDirentsPaste()

### PR DESCRIPTION
**Pull request by haidian6672**

### CODE REVIEW

File: src/filebrowser/file-browser-dialog.cpp [hunk 1]
```diff
@@ -1164,11 +1173,27 @@ void FileBrowserDialog::setFilesToBePasted(bool is_copy, const QStringList &file
...
-        current_path_ == dir_path_to_be_pasted_from_) {
-        seafApplet->warningBox(tr("Cannot paste files from the same folder"), this);
-        return;
+    if (repo_id_to_be_pasted_from_ == repo_.id) {
+        if (current_path_ == dir_path_to_be_pasted_from_) {
+            seafApplet->warningBox(tr("Cannot paste files from the same folder"), this);
+            return;
+        }
+
+        if (file_names_to_be_pasted_.isEmpty()) {
```
Comment (lins05) [inline src/filebrowser/file-browser-dialog.cpp]: No need to check this. If file_names_to_be_pasted_, the paste menu item would not be visible. 

See https://github.com/haiwen/seafile-client/blob/v6.1.0/src/filebrowser/file-table.cpp#L436

File: src/filebrowser/file-browser-dialog.cpp [hunk 2]
```diff
@@ -68,6 +68,15 @@ bool inline findConflict(const QString &name, const QList<SeafDirent> &dirents)
     return found_conflict;
 }
 
+QString appendTrailingSlash(const QString& input) {
+    if (!input.endsWith('/')) {
+        QString ret = input;
+        return ret.append('/');
```
Comment (lins05) [inline src/filebrowser/file-browser-dialog.cpp]: Why do we need this `ret` variable?

The function can be a one-liner like  `input.endsWith("/") ? input : input + "/"`

File: src/filebrowser/file-browser-dialog.cpp [hunk 3]
```diff
@@ -1164,11 +1173,27 @@ void FileBrowserDialog::setFilesToBePasted(bool is_copy, const QStringList &file
...
+
+        if (file_names_to_be_pasted_.isEmpty()) {
+            seafApplet->warningBox(tr("The pasted files do not exist"), this);
+            return;
+        }
+
+        for (const QString& name : file_names_to_be_pasted_) {
+            const QString file_path_to_be_pasted =
+                appendTrailingSlash(::pathJoin(dir_path_to_be_pasted_from_, name));
+            if (appendTrailingSlash(current_path_).startsWith(file_path_to_be_pasted)) {
```
Comment (lins05) [inline src/filebrowser/file-browser-dialog.cpp]: Please add a comment/example here: 

```
// Paste /a/ into /a/b/ is not allowed
```

File: src/filebrowser/file-browser-dialog.cpp (hunk 1 above)
Comment (haidian6672) [reply src/filebrowser/file-browser-dialog.cpp]: But "Ctrl+v" will emit direntPaste() signal.

**Pull request merged**